lib/token.cpp

```diff
@@ -143,10 +143,16 @@ void Token::update_property_info()
                 tokType(eKeyword);
             else if (mTokType != eVariable && mTokType != eFunction && mTokType != eType && mTokType != eKeyword)
                 tokType(eName);
-        } else if (std::isdigit((unsigned char)mStr[0]) || (mStr.length() > 1 && mStr[0] == '-' && std::isdigit((unsigned char)mStr[1])))
-            tokType(eNumber);
-        else if (mStr == "=" || mStr == "<<=" || mStr == ">>=" ||
-                 (mStr.size() == 2U && mStr[1] == '=' && std::strchr("+-*/%&^|", mStr[0])))
+        } else if (std::isdigit((unsigned char)mStr[0]) || (mStr.length() > 1 && mStr[0] == '-' && std::isdigit((unsigned char)mStr[1]))) {
```

Comment:
I think this logic might also exists in simplecpp. We need to keep that in sync or provide in function to get rid of the redundant code. I might be mistaken though.

Reply:
I didn't see any code that is directly related.

Reply:
`simplecpp::Token::flags()` in `simplecpp.h`.

There was talk about sharing this logic with some other dev but I guess that hasn't happened yet. Oops, the PR is actually still open: #4701. Also https://github.com/danmar/simplecpp/pull/285.

Reply:
I see. `101010b` was also discussed in there. This is getting ugly.

Reply:
Ugh, that string-to-number stuff also plays into this. At least I got rid of the bogus calls and added a different conversion function but I haven't removed the code from the `MathLib` conversion functions which does not apply to number literals yet.

Reply:
Hmm. It seems I didn't add anything which is not about number literals (my memory is a mess). If I did it must have been in #4611. But I don't think it make sense to review that.

We should simply get rid of code which makes invalid test cases pass.

Reply:
*If* you re-use any code from that other PR please credit Gerbo via ` Co-authored-by:` at the end of the commit message.